SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialInterestConstraints.h

```diff
@@ -0,0 +1,270 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "CoreMinimal.h"
+
+#include "SpatialInterestConstraints.generated.h"
+
+namespace SpatialGDK
+{
+struct QueryConstraint;
+}
+class USpatialClassInfoManager;
+
+/**
+ * Query data used to configure Query-based Interest.
+ */
+USTRUCT(BlueprintType)
+struct SPATIALGDK_API FQueryData
+{
+	GENERATED_BODY()
+public:
+	FQueryData() = default;
+	~FQueryData() = default;
+
+	/**
+	 * The root constraint associated with the query generated by this component.
+	 */
+	UPROPERTY(BlueprintReadonly, EditDefaultsOnly, Instanced)
+	class UAbstractQueryConstraint* Constraint;
+
+	/**
+	 * The frequency that entities captured by this query should be updated at.
```

Comment:
I think it should be stated that this feature does not work at the moment.

Reply:
👍

Reply:
Would it be better to hide it from the editor interface in the meantime too (by removing the UPROPERTY marker)?

Reply:
I just updated the comment for now (copied from Interest.h plus a not-supported note). That'll make using it straightforward once runtime support is ready.